### PR TITLE
Add Repeat every X seconds extension

### PR DESCRIPTION
**Pull request by arthuro555**
Adds the "Repeat every X seconds" by @VegeTato, an extension that adds convenience abstractions for timers.  
Reviewed and modified to support both objects and scene timers by me.

### CODE REVIEW

**Comment by Bouh** [comment]
The description is empty, it should explain what is for the extension, and how to use it.
See for **Camera shake** extension, there is a great description!
Personally I list what contain the extension, like with **Extended variable support**.

Descriptions for functions must finish with a dot, also function in behavior.

**RepeatXTimes** have a parameter#2 **time** with label: **The time in seconds between each trigger**
**Repeat** have a parameter#2 **timer** with same description, it's ok just unify the parameter name accross the function when the parameters are same.

I hope you have tested the extension first, I trust you ;)

**Comment by arthuro555** [comment]
> The description is empty.  
> unify the parameter name across the function when the parameters are same

Whoops, those are points i actually noticed while reviewing the extension and had fixed, but i think the changes must have been lost without me noticing as GDevelop crashed multiple times while I fixed the issues.

> For authors, use a comma rather than an esperluette &

🤔 Why does this matter?

> I hope you have tested the extension first, I trust you ;)

Well yes that's part of reviewing the extension :p

File: Extensions/RepeatEveryXSeconds.json [hunk 1]
```diff
@@ -0,0 +1,684 @@
...
+              "name": "Behavior",
+              "optional": false,
+              "supplementaryInformation": "RepeatEveryXSeconds::RepeatTimer",
+              "type": "behavior"
+            }
+          ],
+          "objectGroups": []
+        },
+        {
+          "description": "Allows to trigger the recurrent timer X times again.",
```
Comment (4ian) [inline Extensions/RepeatEveryXSeconds.json:589]: "recurrent timer" => "recurring timer" (here and everywhere ;))

File: Extensions/RepeatEveryXSeconds.json [hunk 2]
```diff
@@ -0,0 +1,684 @@
...
+          "label": "The time between each trigger",
+          "description": "",
+          "extraInformation": [],
+          "hidden": false,
+          "name": "TimerLength"
+        },
+        {
+          "value": "-1",
+          "type": "Number",
+          "label": "How many times is the timer allowed to trigger? -1 for forever.",
```
Comment (4ian) [inline Extensions/RepeatEveryXSeconds.json:666]: "How many times should the timer trigger? -1 for forever."

File: Extensions/RepeatEveryXSeconds.json [hunk 3]
```diff
@@ -0,0 +1,684 @@
...
+          "label": "The name of the timer to repeat",
+          "description": "",
+          "extraInformation": [],
+          "hidden": false,
+          "name": "TimerName"
+        },
+        {
+          "value": "1",
+          "type": "Number",
+          "label": "The time between each trigger",
```
Comment (4ian) [inline Extensions/RepeatEveryXSeconds.json:657]: Specify the unit: "The time between each trigger (in seconds)"

File: Extensions/RepeatEveryXSeconds.json [hunk 4]
```diff
@@ -0,0 +1,684 @@
...
+          "optional": false,
+          "supplementaryInformation": "",
+          "type": "expression"
+        }
+      ],
+      "objectGroups": []
+    }
+  ],
+  "eventsBasedBehaviors": [
+    {
```
Comment (4ian) [inline Extensions/RepeatEveryXSeconds.json:299]: Can you explain the advantage of the behavior compared to a set of actions? This avoids to repeat the name of the timer I guess?

File: Extensions/RepeatEveryXSeconds-header.json [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  "fullName": "Repeat every X seconds",
+  "name": "RepeatEveryXSeconds",
+  "version": "0.1.0",
+  "url": "Extensions/RepeatEveryXSeconds.json",
+  "headerUrl": "Extensions/RepeatEveryXSeconds-header.json",
+  "tags": "repeat,timer,time,convenience",
+  "previewIconUrl": "https://resources.gdevelop-app.com/assets/Icons/repeat.svg",
+  "eventsBasedBehaviorsCount": 1,
+  "eventsFunctionsCount": 2,
+  "description": "Convenience functions for timers to trigger a condition every X seconds.",
```
Comment (4ian) [inline Extensions/RepeatEveryXSeconds-header.json:13]: Not "functions" => "conditions, actions and behaviors".

Remember, the term function should not be surfaced outside of an extension, for users the concepts to understand are "conditions", "actions" and "behaviors" :)

**Review by 4ian — APPROVED** [review]
Added a few comments ;) 
Looks good otherwise!

File: Extensions/RepeatEveryXSeconds.json (hunk 4 above)
Comment (arthuro555) [reply Extensions/RepeatEveryXSeconds.json:299]: > This avoids to repeat the name of the timer I guess?

Yes, that's it. If that is not a good idea though i can make those regular actions.

Comment (4ian) [reply Extensions/RepeatEveryXSeconds.json:299]: I'm not sure tbh, the only real risk for now is that behaviors can't be easily renamed in objects.

**Comment by 4ian** [comment]
Thanks @arthuro555 and @VegeTato!

File: Extensions/RepeatEveryXSeconds.json [hunk 6]
```diff
@@ -0,0 +1,684 @@
...
+          "optional": false,
+          "supplementaryInformation": "",
+          "type": "expression"
+        }
+      ],
+      "objectGroups": []
+    },
+    {
+      "description": "Triggers every X seconds X amount of times.",
+      "fullName": "Repeat every X seconds in X number",
```
Comment (4ian) [inline Extensions/RepeatEveryXSeconds.json:127]: @arthuro555  I merged this a bit fast, this is not proper english.  Can you fix that?

**Pull request merged**